lib/src/Uncertainty/Algorithm/Sensitivity/RankSobolSensitivityAlgorithm.cxx

```diff
@@ -0,0 +1,432 @@
+    varianceI_ = computeIndices();
+  }
+  if (0 == aggregatedFirstOrderIndices_.getDimension())
+  {
+    // Compute aggregate indices
+    aggregatedFirstOrderIndices_ = computeAggregatedIndices(varianceI_, referenceVariance_);
+  }
+  // Indices computed
+  return aggregatedFirstOrderIndices_;
+}
+
+
+
+/** Methods of base class that are not yet implemented **/
+
+Sample RankSobolSensitivityAlgorithm::computeIndices(const Sample & sample, Sample & VTi) const
+{
+ throw NotYetImplementedException(HERE) << "In RankSobolSensitivityAlgorithm::computeIndices(const Sample & sample, Sample & VTi) const";
+}
+
+
+SymmetricMatrix RankSobolSensitivityAlgorithm::getSecondOrderIndices(const UnsignedInteger marginalIndex) const
+{
+ throw NotYetImplementedException(HERE) << "In RankSobolSensitivityAlgorithm::getSecondOrderIndices(const UnsignedInteger marginalIndex = 0) const";
+}
+
+Point RankSobolSensitivityAlgorithm::getTotalOrderIndices(const UnsignedInteger marginalIndex) const
+{
+ throw NotYetImplementedException(HERE) << "In RankSobolSensitivityAlgorithm::getTotalOrderIndices(const UnsignedInteger marginalIndex = 0) const";
+}
+
+Interval RankSobolSensitivityAlgorithm::getTotalOrderIndicesInterval() const
+{
+ throw NotYetImplementedException(HERE) << "In RankSobolSensitivityAlgorithm::getTotalOrderIndicesInterval() const";
+}
+
+/** Aggregated total order indices accessor for multivariate samples */
+Point RankSobolSensitivityAlgorithm::getAggregatedTotalOrderIndices() const
+{
+ throw NotYetImplementedException(HERE) << "In RankSobolSensitivityAlgorithm::getAggregatedTotalOrderIndices() const";
+}
+
+/** Whether to use bootstrap or asymptotic distribution */
+Bool RankSobolSensitivityAlgorithm::getUseAsymptoticDistribution() const
+{
+ throw NotYetImplementedException(HERE) << "In RankSobolSensitivityAlgorithm::getUseAsymptoticDistribution() const";
+}
+
+Distribution RankSobolSensitivityAlgorithm::getTotalOrderIndicesDistribution() const
+{
+ throw NotYetImplementedException(HERE) << "In RankSobolSensitivityAlgorithm::getTotalOrderIndicesDistribution() const";
+}
+
+
+/** Method that draw  the sensitivity graph of a fixed marginal */
+Graph RankSobolSensitivityAlgorithm::draw(UnsignedInteger marginalIndex) const 
+{
+ throw NotYetImplementedException(HERE) << "In RankSobolSensitivityAlgorithm::draw(UnsignedInteger marginalIndex) const ";
+}
+
+/* Virtual constructor */
+RankSobolSensitivityAlgorithm * RankSobolSensitivityAlgorithm::clone() const
+{
+  return new RankSobolSensitivityAlgorithm(*this);
+}
+
+/* String converter */
+String RankSobolSensitivityAlgorithm::__repr__() const
+{
+  OSS oss;
+  oss << "class=" << GetClassName()
+      << " " << SobolIndicesAlgorithmImplementation::__repr__();
+  return oss;
+}
+
+/* Method save() stores the object through the StorageManager */
+void RankSobolSensitivityAlgorithm::save(Advocate & adv) const
+{
+  SobolIndicesAlgorithmImplementation::save(adv);
```

Comment:
input/outputDesign_

Reply:
I do not understand this comment. could you please elaborate?

Reply:
you forgot to save the inputDesign_ outputDesign_ attributes